ostfriesen-layout:0.1.0

TobiBeh opened the pull request:
<!--
Thanks for submitting a package! Please read and follow the submission guidelines detailed in the repository's README and check the boxes below. Please name your PR as `name:version` of the submitted package.

If you want to make a PR for something other than a package submission, just delete all this and make a plain PR.
-->

I am submitting
- [x] a new package
- [ ] an update for a package

<!--
Please add a brief description of your package below and explain why you think it is useful to others. If this is an update, please briefly say what changed.
-->

Description: The HS-Emden-Leer-Typst-Template provides a clean, professional template for academic writing at Hochschule Emden/Leer. It features comprehensive document structure with title page, abstract, table of contents, lists of figures/tables, and declaration of independent processing. The template supports multiple authors, company information, multilingual content (English/German), and glossary/abbreviation handling. It streamlines academic formatting to help students produce well-structured, properly formatted theses and reports that follow academic standards.

<!--
These things need to be checked for a new submission to be merged. If you're just submitting an update, you can delete the following section.
-->

I have read and followed the submission guidelines and, in particular, I
- [x] selected a name that isn't the most obvious or canonical name for what the package does
- [x] added a `typst.toml` file with all required keys
- [x] added a `README.md` with documentation for my package
- [x] have chosen a license and added a `LICENSE` file or linked one in my `README.md`
- [x] tested my package locally on my system and it worked
- [x] `exclude`d PDFs or README images, if any, but not the LICENSE

<!--
The following box only needs to be checked for **template** submissions. If you're submitting a package that isn't a template, you can delete the following section. See the guidelines section about licenses in the README for more details.
-->
- [x] ensured that my package is licensed such that users can use and distribute the contents of its template directory without restriction, after modifying them through normal use.

elegaanz commented on packages/preview/ostfriesen-layout/0.1.0/example/main.typ:
```diff
@@ -0,0 +1,62 @@
+// For package use (after publishing):
+// #import "@preview/hs-emden-leer-template:0.1.0": thesis
+
+// For local development and testing:
+#import "../lib/lib.typ": thesis
```
```suggestion
#import "@preview/ostfriesen-layout:0.1.0": thesis
```

elegaanz commented on packages/preview/ostfriesen-layout/0.1.0/README.md:
````diff
@@ -0,0 +1,287 @@
+Just add the following code to your Typst document:
+
+```typst
+#import "@preview/hs-emden-leer-template:0.1.0": thesis
````
```suggestion
#import "@preview/ostfriesen-layout:0.1.0": thesis
```

elegaanz commented on packages/preview/ostfriesen-layout/0.1.0/example/snapshots/thumbnail.png:
I think it would be better to render this image again with a solid white background, otherwise it may render poorly for dark theme users.

elegaanz reviewed (commented):
Thank you for submitting this package. Can you please fix the few problems listed below and the naming warnings reported by CI?

TobiBeh commented:
I updated all user facing content now. And I keep getting warnings for not using kebab case for the functions that are called within the files. Let me know if it's mandatory to change really everything to kebab case or if it's ok that now all outward facing attributes/files are following the naming conventions.

elegaanz commented:
Even if it is not intended to be public, the `create_document` function is. By importing it in `lib.typ` you add it to the scope of this module, which is your package's entry point. I think it would be nice to rename it too for consistency. Another option if you want to make it more explicit that it is not supposed to be called by users is to prefix it with an underscore. Technically, it will still be accessible by users, but it will signal them that this is not a function they are supposed to call themselves (and it will silence the warning in CI). As far as I can tell, this is the only remaining occurrence of a public function with a non-kebab-case name.

TobiBeh commented:
Just went through all files and renamed all function and variable names to match the kebab-case standard. And also fixed some minor issues I found while testing more thoroughly before it finally goes "live".

elegaanz commented:
Thank you!
